python/ray/tests/test_k8s_operator_examples.py

```diff
@@ -0,0 +1,157 @@
+"""Tests launch and teardown of multiple Ray clusters using Kubernetes
+operator."""
+import sys
+import os
+import subprocess
+import tempfile
+import time
+import unittest
+
+import kubernetes
+import pytest
+import yaml
+
+IMAGE_ENV = "KUBERNETES_OPERATOR_TEST_IMAGE"
+IMAGE = os.getenv(IMAGE_ENV, "rayproject/ray:nightly")
+NAMESPACE = "test-k8s-operator-examples"
+
+
+def retry_until_true(f):
+    def f_with_retries(*args, **kwargs):
+        while True:
+            if f(*args, **kwargs):
+                break
+            else:
+                time.sleep(1)
+
+    return f_with_retries
+
+
+@retry_until_true
+def wait_for_pods(n):
+    client = kubernetes.client.CoreV1Api()
+    pods = client.list_namespaced_pod(namespace=NAMESPACE).items
+    # Double-check that the correct image is use.
+    for pod in pods:
+        assert pod.spec.containers[0].image == IMAGE
+    return len(pods) == n
+
+
+@retry_until_true
+def wait_for_logs():
+    """Check if logs indicate presence of 1 node of type "head" and 2 nodes of
+    type "worker"."""
+    cmd = f"kubectl -n {NAMESPACE} logs ray-operator-pod"\
+        "| grep ^example-cluster: | tail -n 100"
+    log_tail = subprocess.check_output(cmd, shell=True).decode()
+    # Normalize log to make this less sensitive to changes in logging output.
+    stripped_log_tail = "".join(
+        [ch.lower() for ch in log_tail if ch.isalnum()])
+    return ("headnode1" in stripped_log_tail) and (
+        "workernodes2" in stripped_log_tail)
+
+
+def operator_configs_directory():
+    here = os.path.realpath(__file__)
+    ray_python_root = os.path.dirname(os.path.dirname(here))
+    relative_path = "autoscaler/kubernetes/operator_configs"
+    return os.path.join(ray_python_root, relative_path)
+
+
+def get_operator_config_path(file_name):
+    return os.path.join(operator_configs_directory(), file_name)
+
+
+def fill_image_field(pod_config):
+    pod_config["spec"]["containers"][0]["image"] = IMAGE
```

Comment:
prefer to leave this inline instead of defining the function for clarity

Reply:
Good call -- done.